### ISSUE TITLE
I001 false positive triggered by `# noqa: F401` on an import within a multi-group block

### ISSUE THREAD

**Issue by feanil** [astral-sh/ruff#24456]
### Summary
Adding a `# noqa: F401` suppression to an import line within a third-party import block
causes ruff to report I001 ("import block is un-sorted or un-formatted") on the entire
block — even though the identical file without the `# noqa: F401` passes cleanly.

## Reproduction

Two files in the same directory, differing only in the presence of `# noqa: F401` on one line:

**`example.py`** (reports I001):
```python
import os
import sys

import ddt
import pymongo
import pytest

# This import must come after the above for reasons.
import django  # noqa: F401
import requests
```

**`example_clean.py`** (passes):
```python
import os
import sys

import ddt
import pymongo
import pytest

# This import must come after the above for reasons.
import django
import requests
```

**`pyproject.toml`**:
```toml
[tool.ruff.lint]
select = ["I", "F"]
```

## Commands

[example_clean.py](https://github.com/user-attachments/files/26520556/example_clean.py)
[example.py](https://github.com/user-attachments/files/26520555/example.py)

```
ruff check --select I,F example.py       # reports I001
ruff check --select I,F example_clean.py # passes
```

## Expected behaviour

Both files should produce the same I001 result. A `# noqa: F401` comment on an import
line should have no effect on import-sorting analysis.

## Actual behaviour

`example.py` reports:

```
I001 [*] Import block is un-sorted or un-formatted
  --> example.py:8:1
   |
 8 | / import os
 9 | | import sys
10 | |
11 | | import ddt
12 | | import pymongo
13 | | import pytest
14 | |
15 | | # This import must come after the above for reasons.
16 | | import django  # noqa: F401
17 | | import requests
   | |_______________^
```

## Ruff version

```
ruff 0.15.7
```

### Version

0.15.7

**Comment by MichaReiser** [comment]
Hi @feanil 

Thanks for the detailed write up. For me, both `example.py` and `example_clean.py` produce the same error:

```
test/24456 on  main [!+?] via 🐍 v3.13.1
❯ uvx ruff@latest check --select I  example.py
[I001] [*] Import block is un-sorted or un-formatted
  --> example.py:1:1
   |
 1 | / import os
 2 | | import sys
 3 | |
 4 | | import ddt
 5 | | import pymongo
 6 | | import pytest
 7 | |
 8 | | # This import must come after the above for reasons.
 9 | | import django  # noqa: F401
10 | | import requests
   | |_______________^
   |
help: Organize imports

Found 1 error.
[*] 1 fixable with the `--fix` option.

test/24456 on  main [!+?] via 🐍 v3.13.1
❯ uvx ruff@latest check --select I  example-clean.py
[I001](https://docs.astral.sh/ruff/rules/unsorted-imports) [*] Import block is un-sorted or un-formatted
  --> example-clean.py:1:1
   |
 1 | / import os
 2 | | import sys
 3 | |
 4 | | import ddt
 5 | | import pymongo
 6 | | import pytest
 7 | |
 8 | | # This import must come after the above for reasons.
 9 | | import django
10 | | import requests
   | |_______________^
   |
help: Organize imports

Found 1 error.
[*] 1 fixable with the `--fix` option.
```

and the files are identical after running `--fix` with the exception of the `noqa` comment:

```
diff example.py example-clean.py
7c7
< import django  # noqa: F401
---
> import django
```

I tried both the latest version (0.15.9) and 0.15.7

**Comment by feanil** [comment]
@MichaReiser thanks for checking. I looked back at the original place where this was happening and I'm not able to re-produce the error, closing for now but I can re-open with more details if I figure out what's going on and it's relevant here.
